src/platform/orchestrator/src/MpiApi.cpp

```diff
@@ -242,19 +235,25 @@ static void MpiSetRequest(const Request& request, Response& response)
                 std::string responsePayload = "\""+ std::to_string(status) + "\"";
 
                 response.m_status = ((status == MPI_OK) ? StatusCode::OK : StatusCode::BAD_REQUEST);
-                response.m_headers[contentLength] = std::to_string(responsePayload.size());
                 response.m_body = responsePayload;
             }
             else
             {
-                OsConfigLogError(PlatformLog::Get(), "No session found for MPI set request: %s", session.c_str());
+                OsConfigLogError(PlatformLog::Get(), "No session found for MpiSet request: %s", session.c_str());
+                response.m_status = StatusCode::BAD_REQUEST;
             }
         }
         else
         {
-            OsConfigLogError(PlatformLog::Get(), "Invalid MPI set request");
+            OsConfigLogError(PlatformLog::Get(), "Invalid MPI set request: %s", request.m_body.c_str());
+            response.m_status = StatusCode::BAD_REQUEST;
         }
     }
+    else
+    {
+        OsConfigLogError(PlatformLog::Get(), "Failed to parse MPI set request body: %s", request.m_body.c_str());
```

Comment:
> OsConfigLogError(PlatformLog::Get(), "Failed to parse MPI set request body: %s", request.m_body.c_str()); [](http://example.com/codeflow?start=7&length=106)

Same here and in the other places. Not allowed to log payload from request or response for Set and Get (all variants) unless full logging is enabled

Reply:
Can do in a follow-up PR